### PR TITLE
Initial commit of Test Demo

### PR DESCRIPTION
**Pull request by CharliePoole**
First commit establishes the directory structure and sets up the VS solutions and projects, which are all building locally.

Next Up: build script and CI

### CODE REVIEW

**Comment by CharliePoole** [comment]
This is now ready to review for merging. There will be more work needed on this but I feel like the features included are ready to merge and then one of us can work on adding more to it.

Implemented:
* Builds under VS2015 using C#, VB and C++, all targeting .NET 4.5
* Builds under VS2017 using C# and targeting .NET 4.5 and .NET Core.

To Do:
* Get the .NET Core demo to run
* Better checking that the results are what they should be, possibly automated

Note that I deliberately didn't add a reference to the adapter, but install it on the fly in the build script. That's so the projects can be used to test various versions of the adapter.

File: solutions/vs2015/CSharpTestDemo/CSharpTestDemo.runsettings [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  <!--<NUnit>
+    <BasePath>D:\Dev\NUnit\nunit3-vs-adapter\demo\NUnitTestDemo\bin\Release</BasePath>
+    <PrivateBinPath>extras;more.extras</PrivateBinPath>
+    <DefaultTimeout>5000</DefaultTimeout>
+    <WorkDirectory>work</WorkDirectory>
+    <InternalTraceLevel>Info</InternalTraceLevel>
+    <RandomSeed>1234567</RandomSeed>
+  </NUnit>-->
+
+</RunSettings>
```
Comment (OsirisTerje) [inline solutions/vs2015/CSharpTestDemo/CSharpTestDemo.runsettings:34]: It would be great if we added all the possible NUnit options to this runsettings.

**Review by OsirisTerje — COMMENTED** [review]
Just merge this in.  
But, should we keep the code for the platforms separate or together with use of defines?   I tend to think they should be separate, it would mean duplication, but easier to use as templates.   We should also add proper usage comments.

**Comment by OsirisTerje** [comment]
Alternatively, we could use different branches for the different platforms.

File: solutions/vs2015/CSharpTestDemo/CSharpTestDemo.runsettings (hunk 1 above)
Comment (CharliePoole) [reply solutions/vs2015/CSharpTestDemo/CSharpTestDemo.runsettings:34]: I thought I had... that is, all the ones it is possible to use under the adapter. There are console options, of course, that wouldn't apply here.

**Comment by CharliePoole** [comment]
@OsirisTerje Well, we only have two platforms that are supported by the demo and much of the .NET core conditional code is unnecessary... somebody decided that the namespace should be different for .NET Core, which seems unnecessary.

I haven't actually done anything on the .NET Core stuff so that could be a new issue here and whoever does it could decide whether to separate the code or keep it common.

We will also want to decide whether other VS versions need to be supported and if we want to convert to any other languages.

**Comment by OsirisTerje** [comment]
We also have the old and the new csproj formats.   About .net core we have both the 1.X and 2.X. , and we have running .net standard SUT projects with both .net core and .net framework.

**Comment by CharliePoole** [comment]
Let's go ahead and create issues for each change we want to make. I'll start with my ToDo items.

**Pull request merged**